libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -945,20 +949,18 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			appendExternalFunctionCall(function, arguments, false);
 			break;
 		}
-		case FunctionType::Kind::ByteArrayPush:
 		case FunctionType::Kind::ArrayPush:
 		{
+			solAssert(function.bound(), "");
 			_functionCall.expression().accept(*this);
 
 			if (function.parameterTypes().size() == 0)
 			{
 				auto paramType = function.returnParameterTypes().at(0);
 				solAssert(paramType, "");
 
-				ArrayType const* arrayType =
-					function.kind() == FunctionType::Kind::ArrayPush ?
-					TypeProvider::array(DataLocation::Storage, paramType) :
-					TypeProvider::bytesStorage();
+				ArrayType const* arrayType = dynamic_cast<ArrayType const*>(function.selfType());
```

Comment:
```suggestion
				auto arrayType = dynamic_cast<ArrayType const*>(function.selfType());
```
Same type twice. Also, at other places in the same file. Changing is optional :)